src-cpp/app/modules/main/module.cpp

```diff
@@ -4,35 +4,33 @@
 #include "module.h"
 #include "singleton.h"
 #include "modules/main/wallet/module.h"
+#include "../shared/section_item.h"
 
-namespace Modules
+namespace Modules::Main
 {
-namespace Main
+Module::Module(std::shared_ptr<Wallets::ServiceInterface> walletsService, QObject* parent): QObject(parent)
 {
-Module::Module(std::shared_ptr<Wallets::ServiceInterface> walletsService)
-{
-    m_controllerPtr = std::make_unique<Controller>();
-    m_viewPtr = std::make_unique<View>();
+    m_controllerPtr = new Controller(this);
+    m_viewPtr = new View(this);
 
     // Submodules
-    m_walletModulePtr = std::make_unique<Modules::Main::Wallet::Module>(walletsService);
+    m_walletModulePtr = new Modules::Main::Wallet::Module(walletsService, this);
 
     m_moduleLoaded = false;
     connect();
 }
 
 void Module::connect()
 {
-    QObject::connect(m_viewPtr.get(), &View::viewLoaded, this, &Module::viewDidLoad);
-    QObject::connect(dynamic_cast<QObject*>(m_walletModulePtr.get()), SIGNAL(loaded()), this, SLOT(walletDidLoad()));
+    QObject::connect(m_viewPtr, &View::viewLoaded, this, &Module::viewDidLoad);
+    QObject::connect(dynamic_cast<QObject*>(m_walletModulePtr), SIGNAL(loaded()), this, SLOT(walletDidLoad()));
```

Comment:
the same as above

Reply:
won't work as IModuleAccess does not inherit from QObject